monetdbe/_cffi/internal.py

```diff
@@ -33,28 +32,27 @@ def result_fetch_numpy(result: monetdbe_result) -> Mapping[str, np.ndarray]:
         name = make_string(rcol.name)
         type_info = monet_c_type_map[rcol.type]
 
+        np_mask = np.ma.nomask  # type: ignore[attr-defined]
         # for non float/int we for now first make a numpy object array which we then convert to the right numpy type
         if type_info.numpy_type.type == np.object_:
-            np_col: np.ndarray = np.array([extract(rcol, r) for r in range(result.nrows)])
+            values = [extract(rcol, r) for r in range(result.nrows)]
+            np_col: np.ndarray = np.array(values)
+            np_mask = np.array([v is None for v in values])
```

Comment:
To me, this seems almost the same except for an extra list comprehension.
`mask = np_col == type_info.null_value` is evaluated in numpy land and should be faster

my guess is you did this to get rid of the `monetdbType` null value field, see comment above.

Reply:
added an issue for this https://github.com/MonetDBSolutions/MonetDBe-Python/issues/168